smartsim/_core/mli/infrastructure/storage/backbone_feature_store.py

```diff
@@ -126,21 +293,26 @@ class OnCreateConsumer(EventBase):
 
     descriptor: str
     """Descriptor of the comm channel exposed by the consumer"""
+    filters: t.List[EventCategory]
+    """The collection of filters indicating messages of interest to this consumer"""
 
-    def __init__(self, descriptor: str) -> None:
+    def __init__(self, descriptor: str, filters: t.Sequence[EventCategory]) -> None:
         """Initialize the OnCreateConsumer event.
 
         :param descriptor: Descriptor of the comm channel exposed by the consumer
+        :param descriptor: Collection of filters indicating messages of interest
         """
         super().__init__(EventCategory.CONSUMER_CREATED, str(uuid.uuid4()))
         self.descriptor = descriptor
+        self.filters = list(filters)
```

Comment:
moved category to be a class attribute on events. this is much nicer because it doesn't couple anything to an enum. new registrations with custom events only need the class to have a unique category name (which could become a problem, but not yet)

filters are populated with the stringified categories.